src/lock_file/satisfiability.rs

```diff
@@ -1060,11 +1060,28 @@ pub(crate) async fn verify_package_platform_satisfiability(
 
                     // Ensure that the record matches the currently selected interpreter.
                     if let Some(python_version) = &record.0.requires_python {
+                        // Strip upperbound as `uv` also does this in the solve step.
+                        // https://github.com/astral-sh/uv/issues/4022
+                        //
+                        // We do this to avoid the case where the locked version is `>=3.6, <3.7`.
+                        // While the python version is `3.10.0` as this is allowed by uv.
+                        let lower_bound_python_version = python_version
+                            // Clone so we can collect it into a new VersionSpecifiers
+                            .clone()
+                            .into_iter()
+                            .filter(|specifier| {
+                                matches!(
```

Comment:
I guess its also possible to specify the python version as, `~=3.6.1`. That will introduce an upper bound. We now completely ignore this bound. Is that correct?

Reply:
If I read this implementation correctly we should ignore all upper bounds:
https://github.com/astral-sh/uv/pull/4086

Reply:
I missed that indeed, I've reimplemented it to use the uv comparison function